protocols/imap/src/main/java/org/apache/james/imap/decode/ImapRequestLineReader.java

```diff
@@ -685,7 +685,28 @@ public static boolean isQuotedSpecial(char chr) {
         return (IdRange[]) merged.toArray(new IdRange[merged.size()]);
     }
 
-    
+    /**
+     * Consume each character until meet the first non-space character then return that
+     *
+     * @return the character
+     * @throws DecodingException
+     */
+    /**
+     * Reads the first non-space character in the current line. This method will continue
+     * to resume if meet space character until meet the non-space character.
+     *
+     * @return The next first non-space character
+     * @throws DecodingException
+     *             If the end-of-stream is reached.
+     */
+    public char nextNonSpaceChar() throws DecodingException {
```

Comment:
Cause the method `nextWordChar` throw exception if the next char is end of line but I still need to catch the end of line character